src/tests/simulation/src/cluster.rs

```diff
@@ -432,27 +441,49 @@ impl Cluster {
         }
         std::env::set_var("RW_META_ADDR", meta_addrs.join(","));
 
+        let mut sql_endpoint = String::new();
+        let mut backend_args = if let Some(sqlite_data_dir) = conf.sqlite_data_dir.as_ref() {
+            sql_endpoint = format!(
+                "sqlite://{}stest-{}.sqlite?mode=rwc",
+                sqlite_data_dir.display(),
+                Uuid::new_v4()
+            );
+            vec!["--backend", "sql", "--sql-endpoint", &sql_endpoint]
+        } else {
+            vec!["--backend", "etcd", "--etcd-endpoints", "etcd:2388"]
+        };
+
+        // FIXME(kwannoel):
+        // Currently we just use the on-disk version,
+        // but it can lead to randomness due to disk io.
```

Comment:
Curious why it leads to randomness due to disk io? I thought the sqlite's IO operations were transparent, so no matter how long an IO op actually takes, the ordering of madsim won't be affected

Reply:
I'm not sure about what you mean by "sqlite's IO operations were transparent". Could you please elaborate?

Reply:
sqlite's IO is not done through an async `file.read()` function because it's written in C++. `tokio` will simply consider run it as a normal async task